tests/test_scalar.py

```diff
@@ -21,6 +21,9 @@ def test_all_quote_styles():
     temp = String("foo", QuoteStyle.BARE)
     assert temp.to_string() == "foo"
 
+    temp = String("foo-bar", QuoteStyle.BARE)
+    assert temp.to_string() == "foo-bar"
```

Comment:
If you know of cases that it will fail or do the wrong thing, can you add tests here for it to show the bad behavior?

Reply:
With this version I don't know of any invalid ones.  I just recently figured out this is what you were asking for, and not allowing dash anywhere.